oak_functions_launcher/tests/integration_test.rs

```diff
@@ -0,0 +1,118 @@
+         [env!("WORKSPACE_ROOT"),"oak_functions_launcher", "key_value_lookup.wasm"].iter().collect()
+    };
+
+    static ref LOOKUP_DATA_PATH: PathBuf = {
+        // WORKSPACE_ROOT is set in .cargo/config.toml.
+         [env!("WORKSPACE_ROOT"),"oak_functions_launcher", "mock_lookup_data"].iter().collect()
+    };
+
+    static ref ENCLAVE_BINARY_PATH: PathBuf = {
+        let oak_functions_linux_fd_bin_path =
+    oak_functions_test_utils::build_rust_crate_linux("oak_functions_linux_fd_bin")
+        .expect("Failed to build oak_functions_linux_fd_bin");
+        PathBuf::from(oak_functions_linux_fd_bin_path)
+    };
+
+}
+
+#[tokio::test]
+async fn test_launcher_looks_up_key() {
+    let params = oak_functions_launcher::instance::native::Params {
+        enclave_binary: ENCLAVE_BINARY_PATH.to_path_buf(),
+    };
+
+    let (launched_instance, connector_handle, _) = oak_functions_launcher::create(
+        Mode::Native(params),
+        LOOKUP_DATA_PATH.to_path_buf(),
+        WASM_PATH.to_path_buf(),
+        1024,
+    )
+    .await
+    .expect("Fail to create launcher");
+
+    let mut client = schema::OakFunctionsAsyncClient::new(connector_handle);
+    let body = b"test_key".to_vec();
+    let invoke_request = InvokeRequest { body: body };
+
+    let response = client
+        .invoke(&invoke_request)
+        .await
+        .expect("Failed to receive response.");
+
+    assert!(response.is_ok());
+    assert_eq!(b"test_value".to_vec(), response.unwrap().body);
+
+    launched_instance
+        .kill()
+        .await
+        .expect("Failed to stop launcher");
+}
+
+#[tokio::test]
+#[ignore]
+async fn test_load_large_lookup_data() {
+    let params = oak_functions_launcher::instance::native::Params {
+        enclave_binary: ENCLAVE_BINARY_PATH.to_path_buf(),
+    };
+
+    let mut rng = rand::thread_rng();
+    // Initialize > 2 GiB of lookup data to hit the proto limits.
+    let lookup_data_size = 2 * (2 << 30);
+    let entry_size = 2 << 10;
+    // The key in the lookup data maps to itself, so an entry has double key size.
+    let mut entries = std::collections::HashMap::new(); // Vec::with_capacity(lookup_data_size / entry_size / 2);
+
+    let mut key_prefix = vec![0u8; (entry_size / 2) - 4];
+    rng.fill_bytes(key_prefix.as_mut_slice());
+
+    for i in 0..(lookup_data_size / entry_size / 2) {
+        let mut n = key_prefix.clone();
+        n.append(&mut format!("{}", i).into_bytes());
+        entries.insert(n.clone(), n);
+    }
+
+    let lookup_data_file = oak_functions_test_utils::write_to_temp_file(
+        &oak_functions_test_utils::serialize_entries(entries),
+    );
+
+    let status = oak_functions_launcher::create(
+        Mode::Native(params),
+        lookup_data_file.path().to_path_buf(),
+        WASM_PATH.to_path_buf(),
+        (entry_size as u32 / 2) + 1024, // Add some margin be on the safe side.
+    )
+    .await;
+
+    // TODO(#3668): Load 2 GiB of lookup data.
+    assert!(status.is_err())
```

Comment:
Why is this erroring right now?

Reply:
I think the key is: 

`thread 'test_load_large_lookup_data' panicked at 'couldn't send lookup data: Status { code: Internal, message: "failed when invoking the request_dispatcher: request channel closed" }', /workspace/oak_functions_launcher/src/lib.rs:116:9
stack backtrace:`

I am not very sure what exactly leads to this error because the stack trace starts with:

`thread 'test_load_large_lookup_data' panicked at 'couldn't convert the message length usize to u32: TryFromIntError(())', oak_channel/src/message.rs:58:14`.

However, I'd like to investigate this in a different PR.